### PR TITLE
Refactor DailySurveyDecorator

### PR DESCRIPTION
**Pull request by hacodeorg**
[PLC-431](https://codedotorg.atlassian.net/browse/PLC-431)
This PR does a few things
1. Refactor DailySurveyDecorator to not extend from SurveyPipelineWorker to make the class simpler.
2. Remove unused empty fields in survey summary report. 
3. Update UI to behave correctly with the removed fields in the report. 
4. Refactor function name and variable name to make their meaning clearer.
5. Update function comments. Remove redundant comments. 

This PR doesn't change functionality of the class.

### CODE REVIEW

File: apps/src/code-studio/pd/workshop_dashboard/reports/local_summer_workshop_daily_survey/results.jsx [hunk 1]
```diff
@@ -111,26 +111,28 @@ export default class Results extends React.Component {
...
-          }
-          facilitatorId={parseInt(facilitator_id, 10)}
-          facilitatorName={this.props.facilitators[facilitator_id]}
-          questions={this.props.facilitatorAverages['questions']}
-          courseName={this.props.courseName}
-          facilitatorResponseCounts={this.props.facilitatorResponseCounts}
-        />
-      </Tab>
-    ));
+    return Object.keys(this.props.facilitators || {}).map(
```
Comment (hacodeorg) [inline apps/src/code-studio/pd/workshop_dashboard/reports/local_summer_workshop_daily_survey/results.jsx]: This line is the only content change, the rest of this file is just whitespace change.

**Review by clareconstantine — APPROVED** [review]
This is a lot clearer and easier to follow- thanks for the comments and renaming in particular!

File: apps/src/code-studio/pd/workshop_dashboard/reports/local_summer_workshop_daily_survey/results.jsx (hunk 1 above)
Comment (islemaster) [reply apps/src/code-studio/pd/workshop_dashboard/reports/local_summer_workshop_daily_survey/results.jsx]: Under what circumstances is the `facilitators` prop falsy?  It's still required according to the proptypes:
```jsx
facilitators: PropTypes.object.isRequired,
```
If you do want to safely omit this prop, perhaps a default value for the prop (e.g. `static defaultProps = ...`) is a better solution than this inline default.

File: dashboard/lib/pd/survey_pipeline/daily_survey_decorator.rb [hunk 2]
```diff
@@ -23,135 +26,108 @@ class DailySurveyDecorator < SurveyPipelineWorker
...
       # Populate entries for result[:this_workshop] and result[:questions]
-      summaries.each do |summary|
+      data[:summaries].each do |summary|
         # Only process summarization for specific workshops and forms.
         workshop_id, form_id, = summary.values_at(:workshop_id, :form_id)
         next unless workshop_id && form_id
 
         # Check if the current user can see specific data
-        next unless data_visible_to_user?(current_user, summary)
+        next unless summary_visible_to_user?(data[:current_user], summary)
```
Comment (islemaster) [inline dashboard/lib/pd/survey_pipeline/daily_survey_decorator.rb:68]: Here's an idea for later - I'd be tempted to pull any knowledge of the current user out of the decorator, and split into two decorators right here, one for the above logic that applies to any user, and one for the below logic that adds summaries.

**Review by islemaster — APPROVED** [review]
Cleanup looks good, and great improvements to the documentation for the decorator!

File: apps/src/code-studio/pd/workshop_dashboard/reports/local_summer_workshop_daily_survey/results.jsx (hunk 1 above)
Comment (hacodeorg) [reply apps/src/code-studio/pd/workshop_dashboard/reports/local_summer_workshop_daily_survey/results.jsx]: `facilitators` field is created only when doing survey roll-up. Not all workshops require doing survey roll-up; an example is CSF deep dive. Previously, `DailySurveyDecorator` always return an empty hash for this field just to please the UI. I remove that unnecessary code, resulting in undefined `facilitators` value on the client side. 

Will use `defaultProps` for fields that only come from survey roll-up.

File: dashboard/lib/pd/survey_pipeline/daily_survey_decorator.rb (hunk 2 above)
Comment (hacodeorg) [reply dashboard/lib/pd/survey_pipeline/daily_survey_decorator.rb:68]: That makes sense. I'll keep that in mind.

**Pull request merged**